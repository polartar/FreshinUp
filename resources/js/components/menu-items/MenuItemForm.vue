<template>
  <v-card style="background-color: #e5e5e5;">
    <v-card-title>
      <h3 class="grey--text">
        Add menu new item
      </h3>
    </v-card-title>
    <v-layout>
      <v-flex
        xs8
        px-2
      >
        <div class="mb-2 text-uppercase grey--text font-weight-bold">
          Item Title
        </div>
        <v-text-field
          v-model="title"
          placeholder="Enter menu item title"
          single-line
          outline
        />
      </v-flex>
      <v-flex
        xs2
        px-2
      >
        <div class="mb-2 text-uppercase grey--text font-weight-bold">
          servings
        </div>
        <v-text-field
          v-model="servings"
          single-line
          outline
        />
      </v-flex>
      <v-flex
        xs2
        px-2
      >
        <div class="mb-2 text-uppercase grey--text font-weight-bold">
          Cost
        </div>
        <v-text-field
          v-model="cost"
          single-line
          outline
        />
      </v-flex>
    </v-layout>
    <v-layout>
      <v-flex
        xs12
        px-2
      >
        <div class="text-uppercase grey--text font-weight-bold">
          Description
        </div>
        <v-textarea
          v-model="description"
          single-line
          outline
        />
      </v-flex>
    </v-layout>
    <v-card-actions class="pa-2 d-flex">
      <div>
        <v-btn
          depressed
          color="grey"
          disabled
          @click="onCancel"
        >
          Cancel
        </v-btn>
        <v-btn
          depressed
          color="primary"
          @click="save"
        >
          Save changes
        </v-btn>
      </div>
    </v-card-actions>
  </v-card>
</template>

<script>
import MapValueKeysToData from '../../mixins/MapValueKeysToData'

export const DEFAULT_MENU_ITEM = {
  title: '',
  description: '',
  servings: '',
  cost: ''
}

/**
 * Menu Item Component
 * @property {String} title
 * @property {String} description
 * @property {Number} servings
 * @property {Number} cost
 */
export default {
  mixins: [MapValueKeysToData],
  data () {
    return {
      ...DEFAULT_MENU_ITEM
    }
  },
  methods: {
    onCancel () {
      this.$emit('cancel')
    }
  }
}
</script>
