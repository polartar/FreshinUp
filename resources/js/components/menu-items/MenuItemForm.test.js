import { mount, shallowMount } from '@vue/test-utils'

import * as Stories from './MenuItemForm.stories'
import Component from './MenuItemForm.vue'

describe('components/menu-items/MenuItemForm', () => {
  describe('Default', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Loading', async () => {
      const wrapper = mount(Stories.Loading())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('WithData', async () => {
      const wrapper = mount(Stories.WithData())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })



  describe('Props & Computed', () => {
    test('isLoading', async () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.isLoading).toBe(false)

      wrapper.setProps({
        isLoading: true
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.isLoading).toBe(true)
    })
  })


  describe('methods', () => {
    test('onCancel()', async () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.onCancel()
      await wrapper.vm.$nextTick()
      expect(wrapper.emitted().cancel).toBeTruthy()
    })
  })
})
