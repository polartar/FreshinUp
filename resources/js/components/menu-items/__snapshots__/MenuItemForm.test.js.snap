// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/menu-items/MenuItemForm Default Default 1`] = `
<div
  class="container white"
>
  <div
    class="v-card v-sheet theme--light"
    style="background-color: rgb(229, 229, 229);"
  >
    <div
      class="v-card__title"
    >
      <h3
        class="grey--text"
      >
        
      Add menu new item
    
      </h3>
    </div>
     
    <div
      class="layout"
    >
      <div
        class="flex xs8 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        Item Title
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  placeholder="Enter menu item title"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs2 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        servings
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs2 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        Cost
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="layout"
    >
      <div
        class="flex xs12 px-2"
      >
        <div
          class="text-uppercase grey--text font-weight-bold"
        >
          
        Description
      
        </div>
         
        <div
          class="v-input v-textarea v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <textarea
                  rows="5"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-card__actions pa-2 d-flex"
    >
      <div>
        <button
          class="v-btn v-btn--disabled v-btn--depressed theme--light grey--text"
          disabled="disabled"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
        Cancel
      
          </div>
        </button>
         
        <button
          class="v-btn v-btn--depressed theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
        Save changes
      
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/menu-items/MenuItemForm Default WithData 1`] = `
<div
  class="container white"
>
  <div
    class="white v-card v-sheet theme--light"
    style="background-color: rgb(229, 229, 229);"
  >
    <div
      class="v-card__title"
    >
      <h3
        class="grey--text"
      >
        
      Add menu new item
    
      </h3>
    </div>
     
    <div
      class="layout"
    >
      <div
        class="flex xs8 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        Item Title
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  placeholder="Enter menu item title"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs2 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        servings
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="flex xs2 px-2"
      >
        <div
          class="mb-2 text-uppercase grey--text font-weight-bold"
        >
          
        Cost
      
        </div>
         
        <div
          class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <input
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <div
                  class="v-messages__wrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="layout"
    >
      <div
        class="flex xs12 px-2"
      >
        <div
          class="text-uppercase grey--text font-weight-bold"
        >
          
        Description
      
        </div>
         
        <div
          class="v-input v-textarea v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-input--is-label-active v-input--is-dirty theme--light"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <textarea
                  rows="5"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-card__actions pa-2 d-flex"
    >
      <div>
        <button
          class="v-btn v-btn--disabled v-btn--depressed theme--light grey--text"
          disabled="disabled"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
        Cancel
      
          </div>
        </button>
         
        <button
          class="v-btn v-btn--depressed theme--light primary"
          type="button"
        >
          <div
            class="v-btn__content"
          >
            
        Save changes
      
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
